hibernate-core/src/main/java/org/hibernate/query/sqm/function/SelfRenderingSqmFunction.java

```diff
@@ -60,7 +62,21 @@ public FunctionRenderingSupport getRenderingSupport() {
 
 		final ArrayList<SqlAstNode> sqlAstArguments = new ArrayList<>( sqmArguments.size() );
 		for ( SqmTypedNode<?> sqmArgument : sqmArguments ) {
-			sqlAstArguments.add( (SqlAstNode) ( (SqmVisitableNode) sqmArgument ).accept( walker ) );
+			if ( sqmArgument instanceof SqmParameter ) {
```

Comment:
What's the reason for this? There are legitimate uses of functions where a tuple should be passed in rather than the expanded list.

Reply:
I changed this to fix `org.hibernate.orm.test.ordered.HqlOrderByIdsTest` where the following query
```
List<Person> persons = entityManager.createQuery(
					"SELECT p " +
							"FROM Person p " +
							"WHERE p.id IN (:ids) " +
							"ORDER BY FIELD(id, :ids) ", Person.class )
					.setParameter( "ids", Arrays.asList( 3L, 1L, 2L ) )
					.getResultList();
```
was failing, when we try to create the `JdbcParameterBindings` in the `SqmUtil#createJdbcParameterBindings` we have an assertion failure (line 233) `assert expansions.size() == bindValues.size() - 1` because for the second one of the `:ids`   `bindValues.size()` is 0

Reply:
by the way not sure that my solution is the correct /better one

Reply:
Hmm, IMO the function should in the rendering expand the tuple manually instead of doing this expansion in the general rendering code.